photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/FixedEffectCoordinate.scala

```diff
@@ -157,8 +157,8 @@ object FixedEffectCoordinate {
    */
   private def score(fixedEffectDataSet: FixedEffectDataSet, fixedEffectModel: FixedEffectModel): KeyValueScore = {
```

Comment:
This private function in `object FixedEffectCoordinate` is exclusively called by `FixedEffectCoordinate.score` in the class `FixedEffectCoordinate`. Is there a reason to not inline it? Same for `updateModel`, in both fixed and random coordinates. With that and `apply` in the `ScoredGameDatum` object, the code gets much simpler:

```scala
 override protected[algorithm] def score(model: DatumScoringModel): KeyValueScore = {

    model match {
      case fixedEffectModel: FixedEffectModel =>
        val modelBroadcast = fixedEffectModel.modelBroadcast
        KeyValueScore(
          dataSet
            .labeledPoints
            .mapValues {
              dataPoint => ScoredGameDatum(dataPoint, modelBroadcast.value.computeScore(dataPoint.features))
            })

      case _ =>
        throw new UnsupportedOperationException(s"Updating scores with model of type ${model.getClass} " +
          s"in ${this.getClass} is not supported!")
    }
  }
```